modules/gdscript/gdscript_tokenizer.cpp

```diff
@@ -820,6 +824,19 @@ GDScriptTokenizer::Token GDScriptTokenizer::number() {
 		}
 	}
 
+	if (need_digits) {
+		// No digits in hex or bin literal.
+		Token error = make_error(vformat(R"(Expected %s digit after "0%c".)", (base == 16 ? "hexadecimal" : "binary"), (base == 16 ? 'x' : 'b')));
+		error.start_column = column;
+		error.leftmost_column = column;
+		error.end_column = column + 1;
+		error.rightmost_column = column + 1;
```

Comment:
Unsure if the full error info is required here, but added it for completeness, unsure what the difference here is to not using it like below

Reply:
I think it is right. Such errors should be reported as a cursor rather than a selection.